packages/fast-development-site-react/src/components/site/menu.tsx

```diff
@@ -86,15 +153,56 @@ class SiteMenu extends React.Component<ISiteMenuProps & IManagedClasses<ISiteNav
         );
     }
 
+    private renderWaffleButton(state: string): JSX.Element {
+        const className: string = `${this.props.managedClasses.siteMenu_button} ${this.props.managedClasses[`siteMenu_button__${state}`]}`;
+
+        return (
+            <button
+                onClick={this.handleMenuVisibilityToggle}
+                aria-label={state}
+                className={className}
+            >
+                <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    viewBox="0 0 32 32"
+                    className={this.props.managedClasses[`siteMenu_buttonGlyph__${state}`]}
+                >
+                    <title>waffle</title>
+                    <path d={waffleGlyph}/>
+                </svg>
+            </button>
+        )
+    }
+
     private handleMenuVisibilityToggle = (): void => {
         this.setState({
             visibility: !this.state.visibility
+        }, (): void => {
+            if (this.state.visibility) {
+                this.navPaneElement.current.style.display = "block";
+            }
+
+            this.navPaneElement.current.addEventListener("animationend", this.handleNavAnimationEnd);
         });
     }
 
-    private getNavigationAttributes(): any {
-        return { "aria-hidden": this.state.visibility };
+    private handleNavAnimationEnd = (e: AnimationEvent) => {
+        const element = this.navPaneElement.current;
+        element.removeEventListener("animationend", this.handleNavAnimationEnd);
+
+        if (!this.state.visibility) {
+            element.style.display = "none";
```

Comment:
instead of setting a style on the element you can add this into the style above and trigger off the `aria-hidden` attribute you've added